### PR TITLE
🌱 Protect well know registry from accidental push during tilt up

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR makes sure that if you are using tilt up without a local registry, you can't accidentally push to well know repository like our staging repo

cc @sbueringer

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Looks good - just a couple of spelling nits.

The behaviour here is that setting this value only matters if there's no default registry.

**Review by sbueringer — COMMENTED** [review]
Thx for following up on this!!

File: Tiltfile [hunk 1]
```diff
@@ -42,8 +42,16 @@ os_arch = str(local("go env GOARCH")).rstrip("\n")
...
-if settings.get("default_registry") != "":
-    default_registry(settings.get("default_registry"))
+usingLocalRegistry = str(local(kubectl_cmd + " get cm -n kube-public local-registry-hosting || true", quiet = True))
+if not usingLocalRegistry:
+    if settings.get("default_registry") == "":
+        fail("default_registry is required when not using a local registry, please add it to your tilt-settings.yaml/json")
+
+    protectedRegistries = ["gcr.io/k8s-staging-cluster-api"]
+    if settings.get("default_registry") in protectedRegistries:
+        fail("current default_registry is protected, tilt cannot push images to it. Please select another default_registry or add '{}' to the allowed_registries list in your tilt-settings.yaml/json".format(settings.get("default_registry")))
```
Comment (sbueringer) [inline Tiltfile:52]:
```suggestion
        fail("current default_registry '{}' is protected, tilt cannot push images to it. Please select another default_registryin your tilt-settings.yaml/json".format(settings.get("default_registry")))
```
last nit

**Comment by sbueringer** [comment]
Sorry last one :)

**Comment by sbueringer** [comment]
Tested locally. All works!

/lgtm
/approve

**Pull request merged**